pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://mesolite.cluster.engflow.com/invocations/default/3aff0ebc-fb10-4003-b983-7caaa0bbb376?testReportRun=10&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on release-25.2 @ [a80059a42d03d5240f6efdad0324376dca17aa59](https://github.com/cockroachdb/cockroach/commits/a80059a42d03d5240f6efdad0324376dca17aa59):


```
     "schema_w0_326",
     "table_w0_347"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER FUNCTION schema_w1_164.udf_w0_175(INT8, INT8) RENAME TO udf_w0_327",
   "CREATE TABLE schema_w1_79.table_w0_328 (col328_w0_329 \"char\" NOT NULL, col328_w0_330 UUID, col328_w0_331 REGROLE NULL, col328_w0_332 TIME NOT NULL, col328_w0_333 BIT(19) NOT NULL, col328_w0_334 TIME NOT NULL, col328_w0_335 REGTYPE, col328_w0_336 STRING COLLATE da_DK NOT NULL, col328_w0_337 GEOGRAPHY, col328_w0_338 TIME, col328_w0_339 \"char\" NULL, col328_w0_340 REGTYPE, col328_w0_341 BIT(20) NOT NULL, col328_w0_342 REGCLASS NULL, col328_w0_343 UUID NOT NULL, col328_w0_344 DATE NULL, col328_w0_345 DATE NOT NULL, col328_w0_346 STRING NOT NULL AS (CASE WHEN col328_w0_335 IS NULL THEN e'\\x12%76My\\x0b8':::STRING ELSE '{':::STRING END) STORED, PRIMARY KEY (col328_w0_346, col328_w0_341 DESC, col328_w0_336 ASC), FAMILY (col328_w0_339), FAMILY (col328_w0_338), FAMILY (col328_w0_343, col328_w0_342), FAMILY (col328_w0_332), FAMILY (col328_w0_333, col328_w0_340, col328_w0_341, col328_w0_329, col328_w0_331), FAMILY (col328_w0_334), FAMILY (col328_w0_337), FAMILY (col328_w0_335), FAMILY (col328_w0_336, col328_w0_345), FAMILY (col328_w0_346), FAMILY (col328_w0_344, col328_w0_330))",
   "CREATE TABLE schema_w0_326.table_w0_347 AS SELECT public.table_w1_73.crdb_internal_mvcc_timestamp AS table_w1_73_0, public.table_w1_38.crdb_internal_mvcc_timestamp AS table_w1_38_0, schema_w1_79.table_w1_137.col137_w1_149 FROM public.table_w1_73, public.table_w1_38, schema_w1_79.table_w1_137 FETCH FIRST 300000 ROWS ONLY"
  ]
 }
}
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3364
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:522
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:610
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 146
Total Schema Statements Succeeded = 146
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 86
Total Transactions Rolled Back = 73
Total Transactions Executed = 159
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_10_of_10/test.outputs/logTestWorkload1080467434
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1080467434
--- FAIL: TestWorkload (13.17s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #161036 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3]
- #160935 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [cannot specify a foreign key update action and an ON UPDATE expression on the same column] [C-test-failure O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59014

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -1498,7 +1498,9 @@ func (og *operationGenerator) createTableAs(ctx context.Context, tx pgx.Tx) (*op
 				}
 				selectStatement.Exprs = append(selectStatement.Exprs, selectExpr)
 
-				if _, exists := uniqueColumnNames[columnNamesForTable[j]]; exists {
+				// Internal columns will always be aliased to unique names, so they can
+				// never generate duplicate column errors.
+				if _, exists := uniqueColumnNames[columnNamesForTable[j]]; exists && !usingInternalColumn {
 					duplicateColumns = true
 				} else {
 					uniqueColumnNames[columnNamesForTable[j]] = true

